Prepare for Renovate with reusable workflows

ghostwriter opened the pull request:
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Adding Renovate to Laminas, Mezzio repositories. https://gist.github.com/weierophinney/49ca8ae2b916d898dd3d4162b26ff41b

- Configure minimum supported PHP version
- Update composer dependencies
- Update to reusable `Continuous Integration` workflow
- Update to reusable `Automatic Releases` workflow

ghostwriter committed:
Configure minimum supported PHP version

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

ghostwriter committed:
Update to reusable continuous-integration workflow

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

ghostwriter committed:
Update to reusable automatic-releases workflow

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

Ocramius reviewed (changes requested):
Requires rolling back to a much smaller `composer.lock` change (just hash change).

Also unsure about target branch: perhaps a new minor would suffice? :thinking:

ghostwriter committed:
Update composer.lock

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

Ocramius reviewed (approved):
Thanks @ghostwriter!

ghostwriter commented:
@Ocramius this one is not ready yet

Ocramius commented:
Branch to be verified by @laminas

Ocramius commented:
CI failures due to launchpad.net being unavailable at this time

weierophinney committed:
qa: remove empty "extra" section from package definition

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: resolve errors raised by Psalm

- Updated Psalm configuration (totallyTyped => errorLevel=1)
- Don't raise errors on usage of Digest adapter (deprecated)
- Unused variables
- Incorrect types

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>
